kornia/contrib/sam/architecture/image_encoder.py

```diff
@@ -0,0 +1,361 @@
+from __future__ import annotations
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from kornia.core import Module, Tensor, pad, zeros
+
+from .common import LayerNorm2d, MLPBlock
+
+
+# This class and its supporting functions below lightly adapted from the ViTDet backbone available at: https://github.com/facebookresearch/detectron2/blob/main/detectron2/modeling/backbone/vit.py # noqa
+class ImageEncoderViT(Module):
```

Comment:
BTW, I remember @edgarriba has implemented a ViT in the contrib previously. Does this mean we shall deprecate that one? Or at least can we reuse some modules like the attention?

Reply:
(just to link) related  #2186